LongWarOfTheChosen/Src/LW_Overhaul/Classes/UIOutpostManagement.uc

```diff
@@ -83,176 +94,347 @@ function OnRadioTowerUpgradeClicked(UIButton Button)
+	}
+
+	// KDM : On the Avenger, both the Resistance overview screen and, this, Haven screen are placed within a 3D movie;
+	// consequently, a 3D camera needs to be specified.
+	if (!class'Utilities_LW'.static.IsOnStrategyMap())
+	{
+		class'UIUtilities'.static.DisplayUI3D(DisplayTag, CameraTag, 0, true);
 	}
-	RegionalInfo.SetHTMLText("<p align=\'RIGHT\'><font size=\'24\' color=\'#fef4cb\'>" $ strRegionalInfo $ "</font></p>");
+
+	// KDM : Container which will hold our UI components : it's invisible
+    MainPanel = Spawn(class 'UIPanel', self);
+	MainPanel.bAnimateOnInit = false;
+	MainPanel.bIsNavigable = false;
+    MainPanel.InitPanel('ListContainer');
+	MainPanel.SetPosition((Movie.UI_RES_X / 2) - (panelW / 2), panelY);
+    
+	// KDM : Background rectangle
+	ListBG = Spawn(class'UIBGBox', MainPanel);
+	ListBG.bAnimateOnInit = false;
+	ListBG.LibID = class'UIUtilities_Controls'.const.MC_X2Background;
+	ListBG.InitBG('ListBG', 0, 0, panelW, panelH);
+    
+	// KDM : Header (includes title, and region sub-title) 
+	ListTitle = Spawn(class'UIX2PanelHeader', MainPanel);
+	ListTitle.bAnimateOnInit = false;
+	ListTitle.bIsNavigable = false;
+	ListTitle.InitPanelHeader('TitleHeader', m_strTitle, Region.GetDisplayName());
+	ListTitle.SetPosition(m_BorderPadding, m_BorderPadding);
+	ListTitle.SetHeaderWidth(panelW - m_BorderPadding * 2);
+    
+    NextY = ListTitle.Y + ListTitle.Height;
+
+    // KDM : Advent strength in the region
+	RegionalInfo = Spawn(class'UIScrollingText', MainPanel);
+	RegionalInfo.bAnimateOnInit = false;
+	RegionalInfo.InitScrollingText('Outpost_RegionalInfo_LW', "", panelW - m_BorderPadding * 2, m_BorderPadding, ListBG.Y + 46.75 + 6);
+	RegionalInfo.SetHTMLText("<p align=\'RIGHT\'><font size=\'24\' color=\'#fef4cb\'>" $ GetAdventStrengthString(Region) $ "</font></p>");
 	RegionalInfo.SetAlpha(67.1875);
 
     // Resistance MECs
     if (Outpost.GetResistanceMecCount() > 0)
     {
-        ResistanceMecs = Spawn(class'UIScrollingText', MainPanel).InitScrollingText('Outpost_ResistanceMecs_LW', "", 500, 453, RegionalInfo.Y + RegionalInfo.Height + 6);
-        kTag = XGParamTag(`XEXPANDCONTEXT.FindTag("XGParam"));
-        kTag.IntValue0 = Outpost.GetResistanceMecCount();
-        strResistanceMecs = `XEXPAND.ExpandString(m_strResistanceMecs);
-        ResistanceMecs.SetHTMLText("<p align=\'RIGHT\'><font size=\'24\' color=\'#fef4cb\'>" $ strResistanceMecs $ "</font></p>");
+		ResistanceMecs = Spawn(class'UIScrollingText', MainPanel);
+		ResistanceMecs.bAnimateOnInit = false;
+		ResistanceMecs.InitScrollingText('Outpost_ResistanceMecs_LW', "", panelW - m_BorderPadding * 2, m_BorderPadding, RegionalInfo.Y + RegionalInfo.Height);
+        ResistanceMecs.SetHTMLText("<p align=\'RIGHT\'><font size=\'24\' color=\'#fef4cb\'>" $ GetResistanceMecString(Outpost) $ "</font></p>");
         ResistanceMecs.SetAlpha(67.1875);
     }
 
-	for (k=0; k< Outpost.prohibitedjobs.length; k++)	
+	for (i=0; i < Outpost.prohibitedjobs.length; i++)	
```

Comment:
If you could fix the spacing when changing existing code, I'd appreciate it. Two-operand operators should have a space either side, so `i = 0` in this case. (There's a technical term for "two-operand operator" but I can't for the life of me remember what it is 😆 )